module_checksum_upgrade/README.rst

```diff
@@ -0,0 +1,83 @@
+.. image:: https://img.shields.io/badge/license-LGPL--3-blue.png
+   :target: https://www.gnu.org/licenses/lgpl
+   :alt: License: LGPL-3
+
+=======================
+Module Checksum Upgrade
+=======================
+
+This addon provides mechanisms to compute sha1 hashes of installed addons,
+and save them in the database. It also provides a method that exploits these
+mechanisms to update a database by upgrading only the modules for which the
+hash has changed since the last successful upgrade.
+
+Installation
+============
+
+There are no specific installation instructions.
+
+Configuration
+=============
+
+There is nothing to configure.
+
+Usage
+=====
+
+The main method provided by this module is ``upgrade_changed_checksum``
+on ``ir.module.module``. It runs a database upgrade for all installed
+modules for which the hash has changed since the last successful
+run of this method. On success it saves the hashes in the database.
+
+The first time this method is invoked after installing the module, it
+runs an upgrade of all modules, because it has not saved the hashes yet.
+This is by design, priviledging safety. Should this be an issue,
+the method ``_save_installed_checksums`` can be invoked in a situation
+where one is sure all modules on disk are installed and up-to-date in the
+database.
```

Comment:
How can you call the method before installing the addon? :thinking:

Reply:
You don't. Installing this module does nothing until  `upgrade_changed_checksum` is called.

Reply:
Oh sorry I understood bad; it's OK then